lib/src/common/model/repos_commits.dart

```diff
@@ -7,119 +7,153 @@ part of github.common;
 
   /// Commit statistics.
-  CommitStats stats;
+  final CommitStats stats;
 
   /// The files changed in this commit.
-  List<CommitFile> files;
-
-  static RepositoryCommit fromJSON(Map<String, dynamic> input) {
+  final List<CommitFile> files;
+
+  RepositoryCommit._({
+    @required this.url,
+    @required this.sha,
+    @required this.htmlUrl,
+    @required this.commentsUrl,
+    @required this.commit,
+    @required this.author,
+    @required this.committer,
+    @required this.stats,
+    @required this.files,
+    @required this.parents,
+  });
+
+  factory RepositoryCommit.fromJSON(Map<String, dynamic> input) {
     if (input == null) return null;
 
-    var commit = RepositoryCommit()
-      ..url = input['url']
-      ..sha = input['sha']
-      ..htmlUrl = input['html_url']
-      ..commentsUrl = input['comments_url']
-      ..commit = GitCommit.fromJSON(input['commit'] as Map<String, dynamic>)
-      ..author = User.fromJson(input['author'] as Map<String, dynamic>)
-      ..committer = User.fromJson(input['committer'] as Map<String, dynamic>)
-      ..stats = CommitStats.fromJSON(input['stats'] as Map<String, dynamic>);
-
-    if (input['parents'] != null) {
-      commit.parents = (input['parents'] as List<dynamic>)
-          .map((parent) => GitCommit.fromJson(parent))
-          .toList();
-    }
-
-    if (input['files'] != null) {
-      commit.files = (input['files'] as List<dynamic>)
-          .map((file) => CommitFile.fromJSON(file))
-          .toList();
-    }
-
-    return commit;
+    final List<GitCommit> parents = input['parents'] != null
+        ? (input['parents'] as List<dynamic>)
+            .map((parent) => GitCommit.fromJson(parent))
+            .toList()
+        : null;
+
+    final List<CommitFile> files = input['files'] != null
+        ? (input['files'] as List<dynamic>)
+            .map((file) => CommitFile.fromJSON(file))
+            .toList()
+        : null;
+
+    return RepositoryCommit._(
+      url: input['url'],
+      sha: input['sha'],
+      htmlUrl: input['html_url'],
+      commentsUrl: input['comments_url'],
+      commit: GitCommit.fromJSON(input['commit'] as Map<String, dynamic>),
+      author: User.fromJson(input['author'] as Map<String, dynamic>),
+      committer: User.fromJson(input['committer'] as Map<String, dynamic>),
+      stats: CommitStats.fromJSON(input['stats'] as Map<String, dynamic>),
+      parents: parents,
+      files: files,
+    );
   }
 }
 
 /// Model class for commit statistics.
 class CommitStats {
   /// Number of Additions.
-  int additions;
+  final int additions;
 
   /// Number of Deletions.
-  int deletions;
+  final int deletions;
 
   /// Total changes.
-  int total;
+  final int total;
+
+  CommitStats._({
+    @required this.additions,
+    @required this.deletions,
+    @required this.total,
+  });
 
-  static CommitStats fromJSON(Map<String, dynamic> input) {
+  factory CommitStats.fromJSON(Map<String, dynamic> input) {
```

Comment:
Changing the static methods to a factory constructor is also a breaking change. Normal instantiation code can remain the same
```
var cs = CommitStats.fromJSON(json); // still works as a constructor without new keyword, or as a static method call
```
but as you have included in this PR, you can no longer pass the method as a tear-off into other methods. For example
```
void some(Function f) {}

fn(CommitStats.fromJSON); // Can't tear-off a factory constructor
```
This is why you had to update many of the convert calls like this
![image](https://user-images.githubusercontent.com/314858/66217919-65889700-e685-11e9-983a-c23bb0b6fa06.png)

Reply:
Again, these static methods/constructors should remain internal. Users shouldn't create these objects by themselves.